Sources/ManageMaintenance.php

```diff
@@ -1248,9 +1248,12 @@ function AdminBoardRecount()
 
 	// Update the latest message of each board.
 	$request = $smcFunc['db_query']('', '
-		SELECT m.id_board, MAX(m.id_msg) AS local_last_msg
+		SELECT m.id_board,
+			(SELECT MAX(id_msg)
+			FROM {db_prefix}messages
+			WHERE approved = {int:is_approved}
+			AND id_board = m.id_board) AS local_last_msg
```

Comment:
I guess you would like todo a literal join,
```suggestion
		SELECT m.id_board,
			(SELECT MAX(id_msg)
			FROM {db_prefix}messages
			WHERE approved = {int:is_approved}
			AND id_board = m.id_board) AS local_last_msg
		FROM {db_prefix}boards AS m
```

got me from a  ~200ms to a 0,5 ms execute time on a ~600k message table WITHOUT changing index.
My suggestion only works when line 
```
FROM {db_prefix}messages AS m
GROUP BY m.id_board
```
get deleted (1256 - 1257).

Reply:
Using your query results in this:
![image](https://user-images.githubusercontent.com/23568484/127750968-9cc06647-2884-4b31-98df-d2453fb73bc2.png)

Two issues....  The first is the 40 second response.  The second (very minor) issue is too many results, though this is easily corrected.

Reply:
without explain you benchmark make no sense,
i guess the index:
```sql
CREATE UNIQUE INDEX smf_messages_id_board ON smf_messages (id_board, id_msg);
```
is not used or not exists.

Reply:
Yes, id_board is heavily used, actually.  And is used in both explain plans.  

As noted earlier, the problem is only with larger boards.  We have seen that behavior changes dramatically ~1M rows.  In mysql only.  

(I have dummied up a large pg db and it appears to be OK...  More testing needed though.)

One comparison explain plan is here, before & after this PR:
![image](https://user-images.githubusercontent.com/23568484/127752230-ca8dfab5-bf07-4df3-bb3a-c237e590bf18.png)

Reply:
my guess is, that your mysql (5.7?) version is to old for the amount of data,
in 8.0 you could use directly lateral join: https://dev.mysql.com/doc/refman/8.0/en/lateral-derived-tables.html
the mention sql from me, is undepend of the amount of data, reason why it's below 0.0005 seconds at a 600k message table.

Reply:
Yes, 5.7.  And yes, I agree mysql8 and pg both handle this MUCH better.